[bug] Provenance subject pkg:npm/hereby/hereby@1.8.3 does not match the package: pkg:npm/hereby@1.8.3

**Describe the bug**

I just set up the nodejs builder to publish my package with provenance; I had to use the main version because the npm version used in the published workflow is too old and fails, but when I do get it running, it appears to fail because the subject and the package do not match.

**To Reproduce**
Steps to reproduce the behavior:

```
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access
npm verb stack Error: Provenance subject pkg:npm/hereby/hereby@1.8.3 does not match the package: pkg:npm/hereby@1.8.3
npm verb stack     at verifyProvenance (/tmp/tmp.FXPIVTdzY6/node_modules/npm/node_modules/libnpmpublish/lib/provenance.js:216:11)
npm verb stack     at async buildMetadata (/tmp/tmp.FXPIVTdzY6/node_modules/npm/node_modules/libnpmpublish/lib/publish.js:181:26)
npm verb stack     at async publish (/tmp/tmp.FXPIVTdzY6/node_modules/npm/node_modules/libnpmpublish/lib/publish.js:45:44)
npm verb stack     at async otplease (/tmp/tmp.FXPIVTdzY6/node_modules/npm/lib/utils/otplease.js:4:12)
npm verb stack     at async Publish.exec (/tmp/tmp.FXPIVTdzY6/node_modules/npm/lib/commands/publish.js:123:7)
npm verb stack     at async module.exports (/tmp/tmp.FXPIVTdzY6/node_modules/npm/lib/cli-entry.js:61:5)
npm verb cwd /home/runner/work/hereby/hereby
npm verb Linux 5.15.0-1041-azure
npm verb node v18.16.1
npm verb npm  v9.8.0
npm ERR! Provenance subject pkg:npm/hereby/hereby@1.8.3 does not match the package: pkg:npm/hereby@1.8.3
npm verb exit 1
npm verb code 1
```

https://github.com/jakebailey/hereby/actions/runs/5483129736/jobs/9989176122#step:3:399

**Expected behavior**

Publish succeeds.

--- a/internal/builders/nodejs/generate-layout.sh
+++ b/internal/builders/nodejs/generate-layout.sh
@@ -32,10 +32,13 @@ set -euo pipefail
 
 # Get the raw package name and scope from the output of `npm pack --json`
 # This name is of the form '<scope>/<package name>'
-raw_package_scope=$(echo "${PACKAGE_NAME:-}" | cut -d'/' -f1)
-raw_package_name=$(echo "${PACKAGE_NAME:-}" | cut -d'/' -f2)
+# NOTE: `cut -s` will suppress output for all fields if the delimiter is
+# not present.
+raw_package_scope=$(echo "${PACKAGE_NAME:-}" | cut -s -d'/' -f1)
+raw_package_name=$(echo "${PACKAGE_NAME:-}" | cut -s -d'/' -f2)
 if [ "${raw_package_name}" == "" ]; then
-    raw_package_name="${raw_package_scope}"
+    # This is a non-scoped package.
+    raw_package_name="${PACKAGE_NAME:-}"
     raw_package_scope=""
 fi
 # package scope (namespace) is URL(percent) encoded.

